atomic_reactor/plugins/post_generate_maven_metadata.py

```diff
@@ -56,6 +67,90 @@ def get_nvr_components(self, nvr_requests):
 
         return components
 
+    def download_sources(self, url_requests):
+        download_queue = []
+        source_url_to_artifacts = {}
+        remote_source_files = []
+        session = util.get_retrying_requests_session()
+
+        for url_request in url_requests:
+            if 'source-url' not in url_request:
+                msg = 'fetch-artifacts-url without source-url is deprecated'
+                self.log.warning(msg)
+                self.log.user_warning(msg)
+                return []
+
+            source_url = url_request['source-url']
+
+            checksums = {algo: url_request[('source-' + algo)] for algo in
+                         hashlib.algorithms_guaranteed
+                         if ('source-' + algo) in url_request}
+
+            checksum_type = list(checksums.keys())[0]
+            artifact = {
+                'url': url_request['url'],
+                'checksum_type': checksum_type,
+                'checksum': url_request[checksum_type],
```

Comment:
url_request has keys like 'source-md , source-sha'   and you are accessing keys 'md, sha'